src/metatrain/share/schema-dataset.json

```diff
@@ -0,0 +1,147 @@
+                            "$ref": "#/$defs/read_from"
+                        },
+                        "file_format": {
+                            "$ref": "#/$defs/file_format"
+                        },
+                        "length_unit": {
+                            "oneOf": [
+                                {
+                                    "type": "null"
+                                },
+                                {
+                                    "type": "string"
+                                }
+                            ]
+                        }
+                    },
+                    "additionalProperties": false
+                }
+            ]
+        },
+        "targets": {
+            "type": "object",
+            "patternProperties": {
+                ".*": {
+                    "oneOf": [
+                        {
+                            "$ref": "#/$defs/read_from"
+                        },
+                        {
+                            "type": "object",
+                            "properties": {
+                                "quantity": {
+                                    "type": "string"
+                                },
+                                "read_from": {
+                                    "$ref": "#/$defs/read_from"
+                                },
+                                "file_format": {
+                                    "oneOf": [
+                                        {
+                                            "$ref": "#/$defs/file_format"
+                                        },
+                                        {
+                                            "type": "object",
+                                            "additionalProperties": true
+                                        }
+                                    ]
+                                },
+                                "key": {
+                                    "type": "string"
+                                },
+                                "unit": {
+                                    "oneOf": [
+                                        {
+                                            "type": "null"
+                                        },
+                                        {
+                                            "type": "string"
+                                        }
+                                    ]
+                                },
+                                "forces": {
+                                    "$ref": "#/$defs/gradient_section"
+                                },
+                                "stress": {
+                                    "$ref": "#/$defs/gradient_section"
+                                },
+                                "virial": {
+                                    "$ref": "#/$defs/gradient_section"
+                                }
```

Comment:
Should we already that `stress` and `virial` are exclusive in the schema?

Also, the gradients are hardcoded. I think this is fine for now...